File: src/main/java/no/novari/mapping/InstanceValueMetadataMappingService.java
package no.novari.mapping;

import no.novari.model.dtos.InstanceValueMetadataDto;
import no.novari.model.entities.InstanceValueMetadata;
import org.springframework.stereotype.Service;

@Service
public class InstanceValueMetadataMappingService {

    public InstanceValueMetadata toEntity(InstanceValueMetadataDto instanceValueMetadataDto) {
        return InstanceValueMetadata
                .builder()
                .displayName(instanceValueMetadataDto.getDisplayName())
                .type(instanceValueMetadataDto.getType())
                .key(instanceValueMetadataDto.getKey())
                .build();
    }

    public InstanceValueMetadataDto toDto(InstanceValueMetadata instanceValueMetadata) {
        return InstanceValueMetadataDto
                .builder()
                .displayName(instanceValueMetadata.getDisplayName())
                .type(instanceValueMetadata.getType())
                .key(instanceValueMetadata.getKey())
                .build();
    }

}

File: src/main/kotlin/no/novari/flyt/discovery/service/mapping/InstanceValueMetadataMappingService.kt
package no.novari.flyt.discovery.service.mapping

import no.novari.flyt.discovery.service.model.dtos.InstanceValueMetadataDto
import no.novari.flyt.discovery.service.model.entities.InstanceValueMetadata
import org.springframework.stereotype.Service

@Service
class InstanceValueMetadataMappingService {
    fun toEntity(instanceValueMetadataDto: InstanceValueMetadataDto): InstanceValueMetadata {
        return InstanceValueMetadata(
            displayName = requireNotNull(instanceValueMetadataDto.displayName),
            type = requireNotNull(instanceValueMetadataDto.type),
            key = requireNotNull(instanceValueMetadataDto.key),
        )
    }

    fun toDto(instanceValueMetadata: InstanceValueMetadata): InstanceValueMetadataDto {
        return InstanceValueMetadataDto(
            displayName = requireNotNull(instanceValueMetadata.displayName),
            type = requireNotNull(instanceValueMetadata.type),
            key = requireNotNull(instanceValueMetadata.key),
        )
    }
}
